draft-ietf-ppm-dap.md

```diff
@@ -1164,14 +1164,9 @@ If obtaining aggregate shares fails, then the leader responds to subsequent HTTP
 GET requests to the collect job URI with an HTTP error status and a problem
 document as described in {{errors}}.
 
-The leader MUST retain a collect job's results until the collector sends an HTTP
-DELETE request to the collect job URI, in which case the leader responds with
-HTTP status 204 No Content.
-
-[OPEN ISSUE: Allow the leader to drop aggregate shares after some reasonable
-amount of time has passed, but it's not clear how to specify that. ACME doesn't
-bother to say anything at all about this when describing how subscribers should
-fetch certificates: https://datatracker.ietf.org/doc/html/rfc8555#section-7.4.2]
+The leader MUST remove a collect job's results when the collector sends an HTTP
+DELETE request to the collect job URI. The leader responds with HTTP status 204
+No Content for requests to a collect job URI whose results have been removed.
```

Comment:
I think this MUST should be rewritten to describe externally observable behavior. For instance:

```suggestion
The collector may send an HTTP DELETE request to the collect job URI, to which
the leader MUST respond with HTTP status 204 No Content. The leader MUST
respond to subsequent requests to the collect job URI with HTTP status 204 No
Content.
```

Note that this also specifies how the leader should respond to the DELETE, which is missing in the existing text. My intuition is that there's no way for other protocol participants to know whether or not the leader's storage still has some aggregate share. The only thing they can see is how the leader responds to a request.

Besides that: this text makes it clear what a leader MUST do if it gets an explicit DELETE, but I think we also need a MAY sentence making it clear that the leader can start respond to the collect URI with HTTP 204 _before_ a collector sends DELETE.

Reply:
OK -- I had left the `MAY` off since it's IMO implied by the updated text, but explicit is usually better than implicit.